### PR TITLE
Add support for unlocking with Touch ID

### PR DESCRIPTION
**Pull request by vantezzen**
### Description
This PR will add support for unlocking Ferdi Lock via macOS Touch ID (as requested in #367). It uses ElectronJS's implementation, thus adding no more dependencies.

On supported systems a new "Use Touch ID to unlock" button will appear on the lock screen. Clicking the button will prompt the user to use TouchID and unlock Ferdi.

This PR will also show a new setting on supported systems that allows the user to disable this feature. The feature is enabled by default though.

### Motivation and Context
For #367.

### Screenshots (if appropriate):
Screenshot 2020-03-02 at 12 56 15
Screenshot 2020-03-02 at 12 56 50

(Needs to be squash-merged due to an unrelated commit that got into the branch)

### CODE REVIEW

File: src/components/auth/Locked.js [hunk 1]
```diff
@@ -19,6 +24,10 @@ const messages = defineMessages({
     id: 'locked.info',
     defaultMessage: '!!!Ferdi is currently locked. Please unlock Ferdi with your password to see your messages.',
   },
+  touchId: {
+    id: 'locked.touchId',
+    defaultMessage: '!!!Use TouchID to unlock',
```
Comment (kytwb) [inline src/components/auth/Locked.js]: Please use **Touch ID** instead of _TouchID_.

File: src/components/auth/Locked.js [hunk 2]
```diff
@@ -64,12 +75,19 @@ export default @observer class Locked extends Component {
     });
   }
 
+  touchIdUnlock() {
+    systemPreferences.promptTouchID('to unlock Ferdi').then(() => {
```
Comment (kytwb) [inline src/components/auth/Locked.js]: Can you share a screenshot of the prompt? To know what message/case fits best.

File: src/containers/settings/EditSettingsScreen.js [hunk 3]
```diff
@@ -88,6 +88,10 @@ const messages = defineMessages({
     id: 'settings.app.form.lockPassword',
     defaultMessage: '!!!Password',
   },
+  useTouchIdToUnlock: {
+    id: 'settings.app.form.useTouchIdToUnlock',
+    defaultMessage: '!!!Allow using TouchID to unlock Ferdi',
```
Comment (kytwb) [inline src/containers/settings/EditSettingsScreen.js]: Please use **Touch ID** instead of _TouchID_.

Comment (kytwb) [reply src/containers/settings/EditSettingsScreen.js]: We can probably drop _Ferdi_ too:
Allow using Touch ID to unlock

**Comment by vantezzen** [comment]
Changed texts to "Touch ID".

The prompt looks something like this:
Screenshot 2020-03-03 at 10 51 00
Of course it will be "Ferdi is..." instead of "Electron is..." in the final release.

**Comment by kytwb** [comment]
@vantezzen Thanks for sharing the screenshot. We can drop the duplicated "to" then in the prompt string: 

> Ferdi is trying ~to~ to unlock Ferdi

Then maybe we can rephrase so it doesn't sound repetitive:

> Ferdi is trying to unlock itself

WDYT?

**Comment by vantezzen** [comment]
Sounds good. I now changed the text to "unlock via Touch ID" and made the text compatible with i18n via 986f3e13.

File: src/containers/settings/EditSettingsScreen.js (hunk 3 above)
Comment (kytwb) [reply src/containers/settings/EditSettingsScreen.js]: @vantezzen This feedback haven't been fully addressed. There is also the part of the feedback regarding not repeating "Ferdi" when not needed that we should start picking up as a habit to enhance the overall copy of the app.

**Comment by vantezzen** [comment]
I don't know about moving the button down, I think it is best to display it above other options as it is most likely the preferred option for anyone using it.
I do think that we should add the shortcut you mentioned to unlock Ferdi - that would be very useful for some users.
I also agree with renaming the "Unlock" button to "Unlock with password" is Touch ID can be used as that button can be confusing otherwise.

**Comment by kytwb** [comment]
Noted for the button position; what do you think about adding a subtle [horizontal rule](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr) with a `or` in the middle (or `OR`, depending on what renders better) to separate the Touch ID button from the password form to mark the separation if Touch ID is available?

Screen Shot 2020-03-03 at 12 14 09 PM

**Comment by vantezzen** [comment]
Screenshot 2020-03-03 at 15 03 29
UI improvements are not implemented.
Screenshot 2020-03-03 at 15 09 13
Unlock screen still looks the same when Touch ID is disabled or not available.

Screenshot 2020-03-03 at 15 06 52
I also added the menu option to unlock with Touch ID.

File: src/lib/Menu.js [hunk 4]
```diff
@@ -833,6 +842,24 @@ export default class FranzMenu {
...
+          systemPreferences.promptTouchID(intl.formatMessage(menuItems.touchIdPrompt)).then(() => {
+            actions.settings.update({
+              type: 'app',
+              data: {
+                locked: false,
+              },
+            });
+          });
+        },
+      });
```
Comment (kytwb) [inline src/lib/Menu.js]:
```suggestion
      }, {
        type: 'separator',
        visible: touchIdEnabled
      });
```

**Comment by kytwb** [comment]
I just added a separator between "Unlock with Touch ID" and "Undo" when applicable.

**Pull request merged**